src/java.base/share/classes/java/lang/foreign/MemoryLayout.java

```diff
@@ -163,6 +163,30 @@
  * long offset2 = (long) offsetHandle.invokeExact(2L); // 16
  * }
  *
+ * <h2 id="record-mapping">Record mapping</h2>
+ * <p>A {@linkplain GroupLayout group layout} can provide mapping capabilities from memory segments to Java
+ * {@linkplain Record Records} by means of matching named elements in the group layout with component names
```

Comment:
This should be a link to the JLS section for records - we do that in other parts of the javadoc - e.g.:

```
Note the use of the <em>try-with-resources</em> construct: this idiom ensures that the off-heap region
 * of memory backing the native segment will be released at the end of the block, according to the semantics described
 * in Section {@jls 14.20.3} of <cite>The Java Language Specification</cite>.
 ```